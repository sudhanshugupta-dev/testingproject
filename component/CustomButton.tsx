import { PropsWithChildren } from 'react';
import { Pressable, Text, StyleSheet, TouchableOpacity } from 'react-native';

export const CustomButton = ({
  children,
  onPress,
}: PropsWithChildren<{ onPress: () => void }>) => {
  return (
    <TouchableOpacity style={styles.button} onPress={onPress}>
      <Text style={styles.text}>{children}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#007AFF',
    padding: 10,
    height: '60%',
    width: '60%',
    borderRadius: 15,
  },
  text: {
    color: 'white',
    textAlign: 'center',
  },
});
