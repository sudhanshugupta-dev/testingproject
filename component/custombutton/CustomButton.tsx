import { PropsWithChildren } from 'react';
import { Pressable, Text, StyleSheet, TouchableOpacity } from 'react-native';
import styles from './Styles';

export const CustomButton = ({
  children,
  onPress,
}: PropsWithChildren<{ onPress: () => void }>) => {
  return (
    <TouchableOpacity style={styles.button} onPress={onPress}>
      <Text style={styles.text}>{children}</Text>
    </TouchableOpacity>
  );
};

