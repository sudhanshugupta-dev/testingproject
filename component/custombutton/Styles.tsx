import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
    button: {
      alignItems: 'center',
      justifyContent: 'center',
      backgroundColor: 'green',
      padding: 10,
      height: '60%',
      width: '60%',
      borderRadius: 20,
    },
    text: {
      color: 'white',
      fontSize: 20,
      textAlign: 'center',
    },
  });

  export default styles;