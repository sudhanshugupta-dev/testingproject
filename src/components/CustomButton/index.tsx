import React from 'react';
import { Pressable, Text, ViewStyle, StyleSheet, ActivityIndicator, View } from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import { useAppTheme } from '../../themes/useTheme';

type Props = { 
  title: string; 
  onPress?: () => void; 
  style?: ViewStyle; 
  disabled?: boolean; 
  testID?: string;
  loading?: boolean;
  variant?: 'primary' | 'secondary' | 'danger';
};

const CustomButton = ({ title, onPress, style, disabled, loading, variant = 'primary', testID }: Props) => {
  const { colors, fonts } = useAppTheme();

  const getGradientColors = () => {
    switch (variant) {
      case 'secondary':
        return [colors.secondary, colors.accent];
      case 'danger':
        return [colors.danger, '#F97316'];
      default:
        return [colors.primary, colors.primaryLight];
    }
  };

  const handlePress = () => {
    if (!disabled && !loading && onPress) {
      onPress();
    }
  };

  return (
    <View style={[styles.container, style]}>
      <Pressable 
        disabled={disabled || loading} 
        testID={testID || 'custom-button'} 
        onPress={handlePress}
        style={({ pressed }) => [
          { 
            opacity: (pressed || disabled || loading) ? 0.8 : 1,
            transform: pressed ? [{ scale: 0.98 }] : [{ scale: 1 }]
          }
        ]}
      >
        <LinearGradient 
          colors={disabled || loading ? ['#D1D5DB', '#9CA3AF'] : getGradientColors()} 
          start={{ x: 0, y: 0 }} 
          end={{ x: 1, y: 0 }} 
          style={[styles.btn, disabled && styles.disabled]}
        >
          {loading ? (
            <ActivityIndicator testID="loading-indicator" color="#ffffff" size="small" />
          ) : (
            <Text style={[
              styles.text, 
              { 
                fontWeight: fonts.weight.bold as any, 
                fontSize: fonts.sizes.large,
                opacity: disabled ? 0.6 : 1
              }
            ]}>
              {title}
            </Text>
          )}
        </LinearGradient>
      </Pressable>
    </View>
  );
};

const styles = StyleSheet.create({ 
  container: {
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 3,
    },
    shadowOpacity: 0.25,
    elevation: 6,
    borderRadius: 16,
  },
  btn: { 
    paddingVertical: 16, 
    paddingHorizontal: 32,
    borderRadius: 16, 
    alignItems: 'center',
    justifyContent: 'center',
    minHeight: 52,
  }, 
  text: { 
    color: '#ffffff',
    letterSpacing: 0.5,
  },
  disabled: {
    opacity: 0.6,
  },
});

export default CustomButton;
