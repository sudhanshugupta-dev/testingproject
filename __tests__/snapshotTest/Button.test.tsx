import React from 'react';
import { render } from '@testing-library/react-native';
import { CustomButton } from '../../component/custombutton/CustomButton';

describe('CustomButton', () => {
  test('matches snapshot with default props', () => {
    const { toJSON } = render(
      <CustomButton onPress={jest.fn()}>Click Me</CustomButton>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
