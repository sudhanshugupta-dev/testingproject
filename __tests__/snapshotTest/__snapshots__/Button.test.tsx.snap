// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton matches snapshot with default props 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "Yellow",
      "borderRadius": 15,
      "height": "60%",
      "justifyContent": "center",
      "opacity": 1,
      "padding": 10,
      "width": "60%",
    }
  }
>
  <Text
    style={
      {
        "color": "white",
        "textAlign": "center",
      }
    }
  >
    Click Me
  </Text>
</View>
`;
